### ISSUE TITLE
Resources in /META-INF/ of executable-fat-jar would be fetched repeatedly.

### ISSUE THREAD

**Issue by QilongZhang** [spring-projects/spring-boot#16536]
When execute a Spring Boot Fat Jar，it's bootstrapped in `AppClassLoader`,  actually switch to `LaunchedURLClassLoader` during refreshing application context. 

As the parent of `LaunchedURLClassLoader` is `AppClassLoader`, it will lead resource in `/META-INF/*` of executable-fat-jar to be loaded repeatedly when invoke `LaunchedURLClassLoader.getResources`:

```java
    public Enumeration<URL> getResources(String name) throws IOException {
        Objects.requireNonNull(name);
        @SuppressWarnings("unchecked")
        Enumeration<URL>[] tmp = (Enumeration<URL>[]) new Enumeration<?>[2];
        if (parent != null) {
            tmp[0] = parent.getResources(name);
        } else {
            tmp[0] = BootLoader.findResources(name);
        }
        tmp[1] = findResources(name);

        return new CompoundEnumeration<>(tmp);
    }
```

it leads our program failed to run！would you fix it?

**Comment by philwebb** [comment]
I'm sorry but I can't understand the issue as it's reported. Could you please provide a sample application that shows the failure.

**Comment by QilongZhang** [comment]
Thanks a lot. I found this bug has been fixed in 1.4.x.  

https://github.com/spring-projects/spring-boot/issues/7449
https://github.com/spring-projects/spring-boot/issues/7595

**Comment by philwebb** [comment]
@QilongZhang 1.4.x is no longer supported, you should upgrade to 2.1.x
